project/src/main/webapp/script.js

```diff
@@ -360,3 +369,29 @@ function addPlaceMarker(map, place) {
     map.setCenter(marker.position);
   });
 }
+
+/**
+ * Called when a user signs in with a Google account.
+ * Updates the global google user, and displays a welcoming messege.
+ */
+function onSignIn(user) {
+  document.getElementById('user-welcome-messeage-container').innerText =
+      "Hello, " + user.getBasicProfile().getName() + "!";
+  googleUser = user;
+}
+
+/** Called when a user signs out of a Google account, updates the screen and the global user. */
+function signOut() {
+  gapi.auth2.getAuthInstance().signOut();
+  document.getElementById('user-welcome-messeage-container').innerText =
+      'You are currently not logged in with a Google account.';
+  googleUser = null;
+}
+
+/** Registers the logged in user, using the registration servlet. */
+function registerUserByToken() {
+  if (!googleUser) {
```

Comment:
If this is already checked before this method is called so maybe this isn't needed.

Reply:
You are right, it's checked twice. I believe in double checking to avoid errors. Do you think it's better to have only one?

Reply:
I would probably use just one since the same is checked right before and right after entering the function, but if you like it it's fine :)